db/special_docs.go

```diff
@@ -43,6 +54,10 @@ func (db *Database) putSpecial(doctype string, docid string, matchRev string, bo
 	if err != nil {
 		return "", base.HTTPErrorf(http.StatusBadRequest, "Invalid expiry: %v", err)
 	}
+
```

Comment:
The current body.getExpiry() doesn't let us differentiate between _exp=0, and _exp not present.  It feels like we need that distinction in this case, to allow users to disable expiry per-doc.

Reply:
Added a bool return value 'present' to indicate that _exp was found in document